### ISSUE TITLE
Feature:  DOCKING

### ISSUE THREAD

**Issue by sedetweiler** [jagenjo/litegraph.js#411]
In products like Substance Designer there are a few QoL improvements because the graphs are so huge and one of the big ones I would love to see here is "docking."  That is similar to minimizing the node, but it parks it on the side of the downstream node and just displays the name.  So, quite similar to the minimize the because it is hanging like a tab from the next node, it is must neater visually and they move together.

**Comment by sedetweiler** [comment]
> Would that be this? Or how? ![image](https://user-images.githubusercontent.com/15611089/258351482-cffa1d6e-766b-47f8-836e-d13dd1e0e20f.png)

Exactly! That would be amazing at cleaning up the graph when we have atomic nodes that require no parameters or additional input.

**Comment by sedetweiler** [comment]
In software like Substance Designer only certain nodes can be docked, and perhaps that is how you can keep this from getting unwieldy.  The docked node acts like part of the parent node when moved about, and this can help so much when you have a lot of little single-function nodes hanging around.  I am currently exposed to this project via ComfyUI and they do not have subgraphs at this time, so the docked node is really something I would love to see there to help with the visual distractions.

**Comment by sefgit** [comment]
collapsable groups would be a nice alternatives.

**Comment by s-light** [comment]
Blender solved this by just having a *input-box* in the component itself - 
as long as there is nothing connected to this input..
![image](https://github.com/jagenjo/litegraph.js/assets/1340319/ccf4fb70-8501-4232-b6ab-1bedd2d8378f)

@sefgit what would be the *visual* / *use-case* difference for you between collapsed group and a sub-graph?

**Comment by sefgit** [comment]
My understanding is that "sub-graph" is to separate source code as modules.
While "grouping codes" is in one module only. Thus collapsable groups might organized nodes neatly. 
More canvas space available for more nodes but still having great overview of whole system.
